### PR TITLE
feat(expr): Add support for dividing intervals by integers/floats/decimals

### PR DESCRIPTION
**Pull request by Graphcalibur**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Add support for dividing intervals by integers/floats/decimals (an error is thrown if a zero division is done)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#2571

### CODE REVIEW

File: src/expr/src/vector_op/arithmetic_op.rs [hunk 1]
```diff
@@ -201,6 +201,22 @@ where
     interval_int_mul::<T2, T1, T3>(r, l)
 }
 
+#[inline(always)]
+pub fn interval_float_div<T1, T2, T3>(l: IntervalUnit, r: T2) -> Result<IntervalUnit>
+where
+    T2: TryInto<OrderedF64> + Debug,
+{
+    l.div_float(r).ok_or(ExprError::NumericOutOfRange)
```
Comment (fuyufjh) [inline src/expr/src/vector_op/arithmetic_op.rs:209]: Actually PG will return a "division by zero" error, but this is acceptable to me.

File: src/expr/src/expr/expr_binary_nonnull.rs [hunk 2]
```diff
@@ -401,6 +401,18 @@ pub fn new_binary_expr(
...
+                    { interval, int64, interval, interval_float_div },
+                    { interval, float32, interval, interval_float_div },
+                    { interval, float64, interval, interval_float_div },
+                    { interval, decimal, interval, interval_float_div },
+                    { int16, interval, interval, float_interval_div },
+                    { int32, interval, interval, float_interval_div },
+                    { int64, interval, interval, float_interval_div },
+                    { float32, interval, interval, float_interval_div },
+                    { float64, interval, interval, float_interval_div },
+                    { decimal, interval, interval, float_interval_div },
```
Comment (fuyufjh) [inline src/expr/src/expr/expr_binary_nonnull.rs:415]: number / interval ?

**Review by neverchanje — COMMENTED** [review]
Pls add some e2e tests as well

File: src/expr/src/expr/expr_binary_nonnull.rs (hunk 2 above)
Comment (Graphcalibur) [reply src/expr/src/expr/expr_binary_nonnull.rs]: They're removed now.

**Review by fuyufjh — APPROVED** [review]

**Pull request merged**